src/main/java/budgetbuddy/logic/commands/transactioncommands/TransactionDeleteCommand.java

```diff
@@ -37,7 +37,7 @@ public TransactionDeleteCommand(Index targetIndex) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
 
-        requireAllNonNull(model, model.getAccountBook());
+        requireAllNonNull(model);
```

Comment:
Also, there seems to be a `removeTransaction` method inside the AccountsManager, so perhaps you should be calling that rather than `model.deleteTransaction`